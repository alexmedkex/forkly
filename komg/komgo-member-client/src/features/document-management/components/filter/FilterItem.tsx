import styled from 'styled-components'
import { blueGrey, violetBlue } from '../../../../styles/colors'
import * as React from 'react'

export interface IFilterItemProps {
  filterKey: string
  title: string
  count: string
  activeKey: string
  onFilter: (key) => void
}

export const FilterItem: React.FC<IFilterItemProps> = (props: IFilterItemProps) => {
  return (
    <ItemWrap
      active={props.filterKey && props.activeKey === props.filterKey}
      onClick={() => props.onFilter(props.filterKey)}
    >
      {props.title}
      <CountWrap>({props.count})</CountWrap>
    </ItemWrap>
  )
}

export const Wrapper = styled.div`
  display: flex;
  flex-wrap: wrap;
  margin-bottom: 30px;
`

export const CountWrap = styled.span`
  margin-left: 4px;
  font-weight: normal;
`

export const ItemWrap =
  styled.span <
  { active: boolean } >
  `
  text-align: center;
  margin: 0 10px;
  cursor: pointer;
  color: ${blueGrey};
  font-weight: bold;
  padding-bottom: 6px;

  &:hover {
    color: black;
  }

  ${props =>
    props.active
      ? `
        color: ${violetBlue}
        border-bottom: 2px solid ${violetBlue};
        &:hover {
          color: ${violetBlue};
        }
      `
      : ``};

`
