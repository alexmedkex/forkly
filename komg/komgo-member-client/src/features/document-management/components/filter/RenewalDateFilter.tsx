import styled from 'styled-components'
import { FilterItem } from './FilterItem'
import * as React from 'react'

import { renewalDateFilterOptions } from '../../utils/filters'

export interface IRenewalDateFilterProps {
  count: Array<{
    key: string
    value: number
  }>
  activeKey: string
  onFilter: (key) => void
}

export interface ItemProps {
  active?: boolean
}

export const RenewalDateFilter: React.FC<IRenewalDateFilterProps> = (props: IRenewalDateFilterProps) => {
  const filterItems = renewalDateFilterOptions.map(filter => {
    const count = props.count ? props.count.find(c => c.key === filter.key) : null

    return {
      title: filter.title,
      filterKey: filter.key,
      count: count ? count.value.toString() : ''
    }
  })

  return (
    <Wrapper>
      {filterItems.map(filter => (
        <FilterItem
          data-test-id={`renewal-date-filter-${filter.filterKey}`}
          key={filter.filterKey}
          {...filter}
          onFilter={props.onFilter}
          activeKey={props.activeKey}
        />
      ))}
    </Wrapper>
  )
}

export const Wrapper = styled.div`
  display: flex;
  flex-wrap: wrap;
  margin-bottom: 30px;
`
