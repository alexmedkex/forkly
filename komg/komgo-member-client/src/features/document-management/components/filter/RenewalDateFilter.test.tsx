import * as renderer from 'react-test-renderer'
import * as React from 'react'

import { RenewalDateFilter, IRenewalDateFilterProps } from './RenewalDateFilter'
import { shallow } from 'enzyme'
import { FilterItem } from './FilterItem'

const defaultProps: IRenewalDateFilterProps = {
  count: [
    {
      key: 'all',
      value: 2
    }
  ],
  activeKey: 'all',
  onFilter: jest.fn()
}

it('should match snapshot', () => {
  const component = renderer.create(<RenewalDateFilter {...defaultProps} />).toJSON()

  expect(component).toMatchSnapshot()
})

it('should render items', () => {
  const wrapper = shallow(<RenewalDateFilter {...defaultProps} />)

  expect(wrapper.find(FilterItem).length).toBe(7)
})
