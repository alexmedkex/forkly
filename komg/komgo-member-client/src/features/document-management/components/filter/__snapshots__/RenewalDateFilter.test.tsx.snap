// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshot 1`] = `
.c2 {
  margin-left: 4px;
  font-weight: normal;
}

.c1 {
  text-align: center;
  margin: 0 10px;
  cursor: pointer;
  color: #5d768f;
  font-weight: bold;
  padding-bottom: 6px;
  color: #5700b5;
  border-bottom: 2px solid #5700b5;
}

.c1:hover {
  color: black;
}

.c1:hover {
  color: #5700b5;
}

.c3 {
  text-align: center;
  margin: 0 10px;
  cursor: pointer;
  color: #5d768f;
  font-weight: bold;
  padding-bottom: 6px;
}

.c3:hover {
  color: black;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  margin-bottom: 30px;
}

<div
  className="c0"
>
  <span
    className="c1"
    onClick={[Function]}
  >
    All
    <span
      className="c2"
    >
      (
      2
      )
    </span>
  </span>
  <span
    className="c3"
    onClick={[Function]}
  >
    KYC renewal due ≤ 30 days
    <span
      className="c2"
    >
      (
      
      )
    </span>
  </span>
  <span
    className="c3"
    onClick={[Function]}
  >
    Due ≤ 60 days
    <span
      className="c2"
    >
      (
      
      )
    </span>
  </span>
  <span
    className="c3"
    onClick={[Function]}
  >
    Due ≤ 90 days
    <span
      className="c2"
    >
      (
      
      )
    </span>
  </span>
  <span
    className="c3"
    onClick={[Function]}
  >
    Overdue ≤ 30 days
    <span
      className="c2"
    >
      (
      
      )
    </span>
  </span>
  <span
    className="c3"
    onClick={[Function]}
  >
    Overdue &gt; 30 days
    <span
      className="c2"
    >
      (
      
      )
    </span>
  </span>
  <span
    className="c3"
    onClick={[Function]}
  >
    No renewal date
    <span
      className="c2"
    >
      (
      
      )
    </span>
  </span>
</div>
`;
